lib/ansible/module_utils/aws/rds.py

```diff
@@ -0,0 +1,178 @@
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+DEFAULT_PORTS = {
+    'aurora': 3306,
+    'mariadb': 3306,
+    'mysql': 3306,
+    'oracle': 1521,
+    'sqlserver': 1433,
+    'postgres': 5432,
+}
+
+DB_ENGINES = [
+    'MySQL',
+    'aurora'
+    'mariadb',
+    'oracle-ee',
+    'oracle-se',
+    'oracle-se1',
+    'oracle-se2',
+    'postgres',
+    'sqlserver-ee',
+    'sqlserver-ex',
+    'sqlserver-se',
+    'sqlserver-web',
+]
+
+LICENSE_MODELS = [
+    'bring-your-own-license',
+    'general-public-license',
+    'license-included',
+    'postgresql-license'
+]
+
+PARAMETER_MAP = {
+    'apply_immediately': 'ApplyImmediately',
+    'backup_retention': 'BackupRetentionPeriod',
+    'backup_window': 'PreferredBackupWindow',
+    'character_set_name': 'CharacterSetName',
+    'cluster': 'DBClusterIdentifer',
+    'db_engine': 'Engine',
+    'db_name': 'DBName',
+    'engine_version': 'EngineVersion',
+    'force_failover': 'ForceFailover',
+    'db_instance_id': 'DBInstanceIdentifier',
+    'instance_type': 'DBInstanceClass',
+    'iops': 'Iops',
+    'license_model': 'LicenseModel',
+    'maint_window': 'PreferredMaintenanceWindow',
+    'multi_zone': 'MultiAZ',
+    'option_group': 'OptionGroupName',
+    'parameter_group': 'DBParameterGroupName',
+    'password': 'MasterUserPassword',
+    'port': 'Port',
+    'publicly_accessible': 'PubliclyAccessible',
+    'security_groups': 'DBSecurityGroups',
+    'size': 'AllocatedStorage',
+    'skip_final_snapshot': 'SkipFinalSnapshot"',
+    'source_instance': 'SourceDBInstanceIdentifier',
+    'snapshot': 'DBSnapshotIdentifier',
+    'storage_type': 'StorageType',
+    'subnet': 'DBSubnetGroupName',
+    'tags': 'Tags',
+    'upgrade': 'AutoMinorVersionUpgrade',
+    'username': 'MasterUsername',
+    'vpc_security_groups': 'VpcSecurityGroupIds',
+    'zone': 'AvailabilityZone',
+}
+
+
+def get_db_instance(conn, instancename):
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        else:
+            raise
+    return response['DBInstances'][0]
+
+
+def instance_to_facts(instance):
+    assert 'DBInstanceIdentifier' in instance, "instance argument was not a valid instance"
+    d = camel_dict_to_snake_dict(instance)
+    return d
+
+
+def get_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            return None
+        else:
+            raise
+    return response['DBSnapshots'][0]
+
+
+def snapshot_to_facts(snapshot):
+    assert 'DBSnapshotIdentifier' in snapshot, "snapshot argument was not a valid snapshot"
```

Comment:
So; I put these in because they pick up situations where the wrong type of dictionary comes in (CamelCase vs snake_case - Parameters vs facts).  They made refactoring easier.  I think later they could maybe go but I'm not sure and I still need them for now.